src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -163,52 +156,829 @@ namespace GroundHeatExchangers {
+
+						} // n
+					} // m
+				} // n1			
+			} // m1
+
+			GFNC( NT ) = ( gFunc * ( coilDiameter / 2.0 ) ) / ( 4 * Pi	* fraction * numTrenches * numCoils );
+			LNTTS( NT ) = tLg;
+
+		} // NT time
+	
+	};
+	//******************************************************************************
+
+	Real64
+	GLHESlinky::nearFieldResponseFunction(
+		int const m,
+		int const n,
+		int const m1,
+		int const n1,
+		Real64 const eta,
+		Real64 const theta,
+		Real64 const t
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR:          Matt Mitchell
+		//       DATE WRITTEN:    February, 2015
+		//       MODIFIED         na
+		//       RE-ENGINEERED    na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Calculates the temperature response of from one near-field point to another
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		// USE STATEMENTS:
+
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+
+		// DERIVED TYPE DEFINITIONS:
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		Real64 distance1;
+		Real64 distance2;
+		Real64 errFunc1;
+		Real64 errFunc2;
+		Real64 sqrtAlphaT;
+		Real64 sqrtDistDepth;
+
+		distance1 = distance( m, n, m1, n1, eta, theta);
+
+		sqrtAlphaT = std::sqrt( diffusivityGround * t );
+
+		if ( ! verticalConfig ) {
+			
+			sqrtDistDepth = std::sqrt( std::pow( distance1, 2.0 ) + 4 * std::pow( coilDepth, 2.0 ) );
+			errFunc1 = std::erfc( 0.5 * distance1 / sqrtAlphaT );
+			errFunc2 = std::erfc( 0.5 * sqrtDistDepth / sqrtAlphaT );
+		
+			return errFunc1 / distance1 - errFunc2 / sqrtDistDepth;
+
+		} else {
+
+			distance2 = distanceToFictRing( m, n, m1, n1, eta, theta );
+		
+			errFunc1 = std::erfc( 0.5 * distance1 / sqrtAlphaT );
+			errFunc2 = std::erfc( 0.5 * distance2 / sqrtAlphaT );
+
+			return errFunc1 / distance1 - errFunc2 / distance2;
+
+		}
+
+	};
+	//******************************************************************************
+
+	Real64
+	GLHESlinky::midFieldResponseFunction(
+	int const m,
+	int const n,
+	int const m1,
+	int const n1,
+	Real64 const t
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR:          Matt Mitchell
+		//       DATE WRITTEN:    February, 2015
+		//       MODIFIED         na
+		//       RE-ENGINEERED    na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Calculates the temperature response of from one mid-field point to another
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		// USE STATEMENTS:
+
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+
+		// DERIVED TYPE DEFINITIONS:
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		Real64 errFunc1;
+		Real64 errFunc2;
+		Real64 sqrtAlphaT;
+		Real64 sqrtDistDepth;
+		Real64 distance;
+
+		sqrtAlphaT = std::sqrt( diffusivityGround * t );
+		
+		distance = distToCenter( m, n, m1, n1 );
+		sqrtDistDepth = std::sqrt( std::pow( distance, 2.0 ) + 4 * std::pow( coilDepth, 2.0 ) );
```

Comment:
Performance note: pow_2( x ) is faster than std::pow( x, 2 ) and should be preferred. Yes, these really add up to matter in some of our profiles. We have pow_2 through pow_9 and a few others in ObjexxFCL/Fmath.hh.